connect/mirror/src/test/java/org/apache/kafka/connect/mirror/DedicatedMirrorIntegrationTest.java

```diff
@@ -62,8 +60,10 @@ public void setup() {
     @AfterEach
     public void teardown() throws Throwable {
         AtomicReference<Throwable> shutdownFailure = new AtomicReference<>();
-        mirrorMakers.forEach((name, mirrorMaker) ->
-            Utils.closeQuietly(mirrorMaker::stop, "MirrorMaker worker '" + name + "'", shutdownFailure)
+        mirrorMakers.forEach((name, mirrorMaker) -> {
+                Utils.closeQuietly(mirrorMaker::stop, "MirrorMaker worker '" + name + "'", shutdownFailure);
+                mirrorMaker.awaitStop();
+            }
```

Comment:
We can optimize a little here, which will hopefully reduce the impact this change has on test time when running multiple MM2 nodes:
```suggestion
        mirrorMakers.forEach((name, mirrorMaker) -> {
                Utils.closeQuietly(mirrorMaker::stop, "MirrorMaker worker '" + name + "'", shutdownFailure);
            }
        );
        mirrorMakers.forEach((name, mirrorMaker) -> {
                    mirrorMaker.awaitStop();
            }
```

Reply:
done in latest commit.